cocos/base/CMakeLists.txt

```diff
@@ -1,18 +1,24 @@
 
 if(APPLE)
-    set(COCOS_BASE_SPECIFIC_SRC
-        base/CCUserDefault-apple.mm
-        base/CCController-apple.mm
-        )
-
+    if(MACOSX)
+        set(COCOS_BASE_SPECIFIC_SRC
+            base/CCUserDefault-apple.mm
+            base/CCController-desktop.cpp
+            )
+    elseif(IOS)
+        set(COCOS_BASE_SPECIFIC_SRC
+            base/CCUserDefault-apple.mm
+            base/CCController-apple.mm
+            )
+    endif()
 elseif(ANDROID)
     set(COCOS_BASE_SPECIFIC_SRC
         base/CCUserDefault-android.cpp
         base/CCController-android.cpp
         )
 elseif(LINUX OR WINDOWS)
     set(COCOS_BASE_SPECIFIC_SRC
-        base/CCController-linux-win32.cpp
+        base/CCController-desktop.cpp
         )
 endif()
```

Comment:
v3 not only support cmake, it also has project files too. So need to modify the project files too.

Reply:
Updated, but I have not tested them with actual builds and hardware. I might be able to do so tomorrow!